### PR TITLE
openzfs-tracking.sh - filter, design, minor changes

### PR DESCRIPTION
**Pull request by gmelikov**
- DataTables library for better usability and status filter
- non-applicable commits are visible
- OPENZFS_PATHS cleaned
- fix bug with path to ZFSONLINUX_EXCEPTIONS
- html code cleanup (styles moved to `<style>` tag)
- intro text

[Example page](https://gmelikov.ru/zfs.html)

### CODE REVIEW

File: scripts/openzfs-tracking.sh [hunk 1]
```diff
@@ -84,22 +98,102 @@ while getopts 'hd:e:' OPTION; do
...
 </head>
 <body>
-<table align="center" width="80%" border="0">
-<tr bgcolor='#aaaaaa'>
-  <th colspan='4'>OpenZFS Commit Tracking</th>
-</tr>
-<tr bgcolor='#dddddd'>
+<h1 align='center'>OpenZFS Commit Tracking</h1>
+<div id="intro">
+This page is updated regularly and shows a list of OpenZFS commits and their status in regard to the ZFS on Linux master branch. See <a href="https://github.com/zfsonlinux/zfs/wiki/OpenZFS-Patches">wiki</a> for more information about OpenZFS patches.
```
Comment (dinatale2) [inline scripts/openzfs-tracking.sh]: I don't think you need to explain that this page is updated on a regular basis. We can add text to that effect when we link it on a wiki page.

File: scripts/openzfs-tracking.sh [hunk 2]
```diff
@@ -84,22 +98,102 @@ while getopts 'hd:e:' OPTION; do
...
+<script		  src="https://code.jquery.com/jquery-1.12.4.min.js"
+			  integrity="sha256-ZosEbRLbNQzLpnKIkEdrPv7lOy9C27hHQ+Xp8a4MxAQ="
+			  crossorigin="anonymous"></script>
+<link rel="stylesheet" type="text/css" href="https://cdn.datatables.net/1.10.13/css/jquery.dataTables.min.css">
+<script type="text/javascript" language="javascript" src="https://cdn.datatables.net/1.10.13/js/jquery.dataTables.min.js"></script>
+
+<script type="text/javascript">
+\$(document).ready(function() {
+	\$('#maintable').DataTable( {
+		 "lengthMenu": [[50, 100, 500, -1], [50, 100, 500, "All"]],
```
Comment (dinatale2) [inline scripts/openzfs-tracking.sh]: I think this should default to "All". I'm not 100% sure if you need it honestly.

File: scripts/openzfs-tracking.sh (hunk 1 above)
Comment (gmelikov) [inline scripts/openzfs-tracking.sh]: I added this description (and link to openzfs-tracking.sh) for new contributors. Link to openzfs-tracking.sh can help with understanding how table was generated.

File: scripts/openzfs-tracking.sh [hunk 3]
```diff
@@ -84,22 +98,102 @@ while getopts 'hd:e:' OPTION; do
...
+<div id="f_date">Last Update: $DATE by <a href="https://github.com/zfsonlinux/zfs-buildbot/blob/master/scripts/openzfs-tracking.sh">openzfs-tracking.sh</a></div>
+<table id="maintable" class="display">
+<thead>
+<tr>
   <th>OpenZFS Issue</th>
   <th>OpenZFS Commit</th>
   <th>Linux Commit</th>
-  <th>Description</th>
+  <th class='td_text'>Description</th>
+  <th>Status</th>
```
Comment (gmelikov) [inline scripts/openzfs-tracking.sh]: I can't say that it must be hidden, it's easier personally for me to see the status. We can add any comment to this column later.

I can hide it and add a legend, if it's better. Yes, it's used for filter.

File: scripts/openzfs-tracking.sh [hunk 4]
```diff
@@ -84,22 +98,102 @@ while getopts 'hd:e:' OPTION; do
...
+						var val = $.fn.dataTable.util.escapeRegex(
+							\$(this).val()
+						);
+
+						column
+							.search( val ? '^'+val+'$' : '', true, false )
+							.draw();
+					} );
+
+				column.data().unique().sort().each( function ( d, j ) {
```
Comment (gmelikov) [inline scripts/openzfs-tracking.sh]: I think creating checkboxes is real. If everyone wants checkboxes, i can dig into it later.

**Comment by dinatale2** [comment]
Mind regenerating your sample page?

**Comment by gmelikov** [comment]
@dinatale2 link was in the first message - [example](https://gmelikov.ru/zfs.html).

**Comment by behlendorf** [comment]
Merged!  Sorry about the delay, it was a long holiday weekend.  Regarding the pagination functionality I think that's a perfectly reasonable thing to keep.  I suspect I'd usually want to see everything but I can understand how other peoples workflow might be different.  Particularly as the number of OpenZFS commits increases.

**Comment by gmelikov** [comment]
@behlendorf thanks! there are some OpenZFS commits which, i think, we can add to exceptions, i'll make new PR and add pagination with them next time (with settings to show all records by default).

**Comment by behlendorf** [comment]
@gmelikov that's be great, looking forward to the PR.  I just added a few additional exceptions myself with commit 384194fb7188305a323d5fd8dc249966e0f20332 but I haven't carefully gone through all the commits so I can easily believe there are more.

**Pull request merged**